pkg/totalaccountwatcher/totalaccountwatcher.go

```diff
@@ -0,0 +1,155 @@
+
+var log = logf.Log.WithName("aws-account-operator")
+
+type totalAccountWatcher struct {
+	watchInterval time.Duration
+	AwsClient     awsclient.Client
+	client        client.Client
+	Total         int
+}
+
+// Initialize creates a global instance of the TotalAccountWatcher
+func Initialize(client client.Client, watchInterval time.Duration) {
+	log.Info("Initializing the totalAccountWatcher")
+
+	AwsClient, err := awsclient.GetAWSClient(client, awsclient.NewAwsClientInput{
+		SecretName: AwsSecretName,
+		NameSpace:  awsv1alpha1.AccountCrNamespace,
+		AwsRegion:  "us-east-1",
+	})
+
+	if err != nil {
+		log.Error(err, "Failed to get AwsClient")
+		return
+	}
+
+	TotalAccountWatcher = NewTotalAccountWatcher(client, AwsClient, watchInterval)
+	TotalAccountWatcher.UpdateTotalAccounts(log)
+}
+
+// NewTotalAccountWatcher returns a new instance of the TotalAccountWatcher interface
+func NewTotalAccountWatcher(client client.Client, AwsClient awsclient.Client, watchInterval time.Duration) *totalAccountWatcher {
+	return &totalAccountWatcher{
+		watchInterval: watchInterval,
+		AwsClient:     AwsClient,
+		client:        client,
+	}
+}
+
+// TotalAccountWatcher will trigger AwsLimitUpdate every `scanInternal` and only stop if the operator is killed or a
+// message is sent on the stopCh
+func (s *totalAccountWatcher) Start(log logr.Logger, stopCh <-chan struct{}) {
+	log.Info("Starting the totalAccountWatcher")
+	for {
+		select {
+		case <-time.After(s.watchInterval):
+			log.Info("totalAccountWatcher: updating a")
+			err := s.UpdateTotalAccounts(log)
+			if err != nil {
+				log.Error(err, "totalAccountWatcher not started, awsLimit won't be updated")
+			}
+		case <-stopCh:
+			log.Info("Stopping the totalAccountWatcher")
+			break
+		}
+	}
+}
+
+// AwsLimitUpdate will update the TotalAccountWatcher's total field
+func (s *totalAccountWatcher) UpdateTotalAccounts(log logr.Logger) error {
+
+	// Reset cache
+	secretList := &corev1.SecretList{}
+	secret := &corev1.Secret{}
+
+	listOps := &client1.ListOptions{Namespace: awsv1alpha1.AccountCrNamespace}
+	if err := TotalAccountWatcher.client.List(context.TODO(), listOps, secretList); err != nil {
+		log.Error(err, fmt.Sprintf("Unable to list secrets in namespace %s", awsv1alpha1.AccountCrNamespace))
+	}
+
+	if err := TotalAccountWatcher.client.Get(context.TODO(), types.NamespacedName{Name: AwsSecretName, Namespace: "aws-account-operator"}, secret); err != nil {
+		log.Error(err, fmt.Sprintf("Unable to get secrets in namespace %s", awsv1alpha1.AccountCrNamespace))
+	}
+
+	accountTotal, err := TotalAwsAccounts()
+	if err != nil {
+		log.Error(err, "Failed to get AWS account total from AWS api")
+	}
+
+	TotalAccountWatcher.Total = accountTotal
+	log.Info(fmt.Sprintf("Total AWS accounts: %d", TotalAccountWatcher.Total))
+
+	gciOut, err := TotalAccountWatcher.AwsClient.GetCallerIdentity(&sts.GetCallerIdentityInput{})
+	log.Info(fmt.Sprintf("Caller identity: %+v", gciOut))
```

Comment:
What is this for? I think this might be an unnecessary copy-paste?

Reply:
This was a debug log that I kept in. Can be removed.